client/executor/src/wasm_runtime.rs

```diff
@@ -71,6 +74,9 @@ struct VersionedRuntime {
 	module: Arc<dyn WasmModule>,
 	/// Runtime version according to `Core_version` if any.
 	version: Option<RuntimeVersion>,
+
+	// TODO: Remove this once the legacy instance reuse instantiation strategy
+	//       for `wasmtime` is gone, as this only makes sense with that particular strategy.
```

Comment:
Most likely yes (assuming we care about its performance here; even with the caching it's probably going to be too slow to use for anything else other than testing and/or as a backup executor in case of a disaster), although 1) for `wasmi` we just should just essentially add something like this deeper in the `sc-executor-wasmi` itself (it makes more sense for every backend to decide for itself as to what's the most efficient way to reuse it), and 2) the interface's probably going to change anyway.